File: Graphs/graph-bfs.py
# BFS - Graph traversal
# Time: O(V + E)
# Space: O(V)

from collections import deque

def bfs(graph, start):
    visited = set([start])
    queue = deque([start])

    while queue:
        node = queue.popleft()
        print(node)  # or process node
        for neighbor in graph[node]:
            if neighbor not in visited:
                visited.add(neighbor)
                queue.append(neighbor)

File: Graphs/graph-bfs.js
// BFS - Graph traversal
// Time: O(V + E)
// Space: O(V)

function bfs(graph, start) {
    const visited = new Set([start]);
    const queue = [start];

    while (queue.length > 0) {
        const node = queue.shift();
        console.log(node); // or process node

        for (const neighbor of graph[node] || []) {
            if (!visited.has(neighbor)) {
                visited.add(neighbor);
                queue.push(neighbor);
            }
        }
    }
}
